tests/reframe/eessi_utils/hooks.py

```diff
@@ -0,0 +1,26 @@
+import reframe as rfm
+import eessi_utils.utils as utils
+
+def skip_cpu_test_on_gpu_nodes(test: rfm.RegressionTest):
+    '''Skip test if GPUs are present, but no CUDA is required'''
+    skip = ( utils.is_gpu_present(test) and not utils.is_cuda_required(test) )
+    if skip:
+        print("GPU is present on this partition, skipping CPU-based test")
+        test.skip_if(True)
+
```

Comment:
Maybe we can use the logic from flex-alloc-nodes to skip tests that are too large for the current system. Might not work well on PBS though. 

Logic can be taken from here https://github.com/eth-cscs/reframe/blob/8ecb01eb742cfa61f9ef320ed3c5af9d9908dece/reframe/core/schedulers/__init__.py#L412

Reply:
Scratch that: we want to tag, because for the CI on the build nodes, we _only_ want to run the single node tests anyway. It doesn't make sense to already spend resources on multi-node tests at this stage.